src/content/en/2021/caching.md

```diff
@@ -17,4 +20,425 @@ featured_stat_3: TODO
+
+When we compare the mobile data from 2020 to 2021, we notice a 5.1% increase in cacheable HTML. This tells us we may be moving towards better usage of our CDNs to cache HTML pages, like those rendered by an SSR, that tend to change less frequently. Pages are typically generated by Server-Side Rendered (SSR) applications if the content of a particular web page doesn't change frequently. The url can potentially serve the same HTML for weeks or even months, making that content highly cacheable.
+
+Taking a look at the Median TTL across all resource types, we see that even if we are caching similar percentage in total, there is a much lengthier cache for mobile, particularly for HTML, audio and video.
+
+<figure>
+  <table>
+    <thead>
+      <tr>
+        <th colspan="6" scope="col">Median TTL (in days)</th>
+      </tr>
+      <tr>
+        <th scope="col">Type</th>
+        <th scope="col">Desktop</th>
+        <th scope="col">Mobile</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td>Text</td>
+        <td class="numeric">0.2</td>
+        <td class="numeric">0.2</td>
+      </tr>
+      <tr>
+        <td>XML</td>
+        <td class="numeric">1</td>
+        <td class="numeric">1</td>
+      </tr>
+      <tr>
+        <td>Other</td>
+        <td class="numeric">1</td>
+        <td class="numeric">1</td>
+      </tr>
+      <tr>
+        <td>Video</td>
+        <td class="numeric">4</td>
+        <td class="numeric">8</td>
+      </tr>
+      <tr>
+        <td>HTML</td>
+        <td class="numeric">3</td>
+        <td class="numeric">14</td>
+      </tr>
+      <tr>
+        <td>Audio</td>
+        <td class="numeric">0.2</td>
+        <td class="numeric">30</td>
+      </tr>
+      <tr>
+        <td>CSS</td>
+        <td class="numeric">30</td>
+        <td class="numeric">30</td>
+      </tr>
+      <tr>
+        <td>Image</td>
+        <td class="numeric">30</td>
+        <td class="numeric">30</td>
+      </tr>
+      <tr>
+        <td>Script</td>
+        <td class="numeric">30</td>
+        <td class="numeric">30</td>
+      </tr>
+      <tr>
+        <td>Font</td>
+        <td class="numeric">365</td>
+        <td class="numeric">365</td>
+      </tr>
+    </tbody>
+  </table>
+  <figcaption>TODO: Caption the table above</figcaption>
+</figure>
+
+When we look at the entirety of cacheable and non-cacheable resources,
+
+[TODO add a figure that shows Cacheable vs non-cacheable response similar to https://20211128t100318-dot-webalmanac.uk.r.appspot.com/static/images/2020/caching/cacheable-and-non-cacheable.png]
+
+## How do cache TTLs compare to resource age?
+Between 1st and 3rd party resources:  
+- Images have the same resource age of 2 years. 
+- The majority of Video resources is also between 8-52 weeks
+
+Fonts for 3rd party are cached the most between 8-52 weeks at 72.4%, but interestingly for 1st party, the largest resource age grouping is evenly split between 8-52 weeks and over 2 years.
+
```

Comment:
We see that Images and Videos maintained the same average age whether from 1st or 3rd party resources. Images consistently had the same resource age of 2 years, while most Video resources were between 8-52 weeks old.

Breaking down the other types of content, we discovered Fonts for 3rd parties are cached the most between 8-52 weeks at 72.4%. However, for 1st party the largest resource age groups is evenly split between 8-52 weeks and over 2 years- quite a large variance.
We see similar results for Audio and Scripts where the majority of 1st party are between 8-52 weeks old while for 3rd party they are between 1-7 weeks. 

Audio was the most highly cached resource across both 1st and 3rd parties. However, the resource age varied greatly between first party (averaging 8-52 weeks) and third party, at only 1-7 weeks. Audio resources in first party situations tend to be updated less frequently (why?), so third parties may be capitalizing on a caching opportunity by offering fresher resources. 



For CSS, 1st party the larger group is 8-52 weeks while the larger group for 3rd party with 51.8% is with less than a week.

Finally, HTML has the largest 1st party group served with less than a week with 42.7% and third party's largest group is between 1-7 weeks with 43.1%. 

Considerations after reviewing this data: 
- The freshest content for 1st party is HTML while for 3rd party it is CSS. 
- The most stale content for both 1st and 3rd party is images.

This data shows how 1st parties have prioritized refreshing HTML content which usually holds the link to JS and CSS files, while 3rd party providers that are mostly CSS and script-driven, like browser extensions, have prioritized keeping their CSS up-to-date.

Mobile resources with a cache TTL that was considered too short compared to its content age have seen an improvement since 2020. 60.2% in 2020 but now 54% in 2021.

{{ figure_markup(
  caption="TODO",
  content="54%",
  classes="big-number",
  sheets_gid="768623684",
  sql_file="todo.sql"
) }}